src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,116 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final int MAXIMUM_CAPACITY = 1 << 30;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (table == null) {
+            initialMyHash();
+        }
+        if (size >= threshold) {
```

Comment:
you don't need to create a separate field for this
You can calculate where you need.
```suggestion
        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
```